plugins/admissions/loader.py

```diff
@@ -354,29 +355,28 @@ def load_bed_status():
         Q_GET_ALL_BED_STATUS
     )
 
+    mrns = [i["Local_Patient_Identifier"] for i in status]
+    mrn_to_patient = find_patients_from_mrns(mrns)
+
+    for mrn in mrns:
+        if mrn not in mrn_to_patient:
+            mrn_to_patient[mrn] = create_rfh_patient_from_hospital_number(
+                mrn, InfectionService
+            )
+
     with transaction.atomic():
 
         BedStatus.objects.all().delete()
 
         for bed_data in status:
-            bed_status = BedStatus()
+            patient = mrn_to_patient.get(bed_data["Local_Patient_Identifier"])
+            if not patient:
```

Comment:
I think what 'if not patient` really means is `if bed_data["Local_Patient_Identifier"] == None || bed_data["Local_Patient_Identifier"] == '' || bed_data["Local_Patient_Identifier"].lstrip('0') == '' ` but done implicitly by another function in another file that throws away data it doesn't like? 

I predict that won't be immediately obvious in 6months - as a minimum we should explain how it happens